riscv-executor/src/large_field/mod.rs

```diff
@@ -0,0 +1,1363 @@
+//! A specialized executor for our RISC-V assembly that can speedup witgen and
+//! help with making partition decisions.
+//!
+//! WARNING: the general witness generation/execution code over the polynomial
+//! constraints try to ensure the determinism of the instructions. If we bypass
+//! much of witness generation using the present module, we lose the
+//! non-determinism verification.
+//!
+//! TODO: perform determinism verification for each instruction independently
+//! from execution.
+
```

Comment:
Most of the code here didn't change, was just moved to the `large_field` module.
There are some small changes I'll point out.